### DISCUSSION TITLE
0.5.2 Release Timeline

### DISCUSSION THREAD

**Discussion by isaaccorley** [torchgeo/torchgeo#1909] [Announcements]
We're planning a 0.5.2 release next week. This will include all bug fixes (no new features or API changes) in [0.5.2](https://github.com/microsoft/torchgeo/milestone/17).

If you have any bugs you would like to see fixed, now is the time to submit PRs!

**Comment by adamjstewart** [comment]
# TorchGeo 0.5.2 Release Notes

This is a bugfix release. There are no new features or API changes with respect to the 0.5.1 release. 

This release contains a number of important fixes to reproducibility and determinism. All users are recommended to upgrade to 0.5.2 if they want to ensure the reproducibility of their work. 

TorchGeo has always supported Python 3.12, but this is now officially tested!

## Dependencies

* Test TorchGeo support for Python 3.12 (#1837)
* lightly 1.4.26 is incompatible with smp (#1824, #1825)
* Add dev container to support Github Codespaces development (#1085)

## Datamodules

* L7 Irish previously used a nondeterministic train/val/test split. This is now fixed (#1899, #1908)
* L8 Biome previously used a nondeterministic train/val/test split. This is now fixed (#1899, #1908)
* Tropical Cyclone previously used a nondeterministic train/val/test split. This is now fixed (#1839)
* SEN12MS previously used a nondeterministic train/val/test split. This is now fixed (#1839)

## Datasets

* RasterDataset: clarify documentation of is_image and dtype (#1811)
* GeoDataset previously used a nondeterministic train/val/test split. This is now fixed (#1899, #1908)
* xView2 previously used a nondeterministic order. This is now fixed (#1918)
* HuggingFace: use stable download URLs (#1916)
* GitLab: use stable download URLs (#1917)
* Deep Globe Land Cover: document download steps (#1797, #1921)
* PASTIS: fix default folds (#1810)
* SustainBench Crop Yield: fix download support (#1753, #1755)
* SustainBench Crop Yield: eager data loading (#1754, #1756)

## Models

* HuggingFace: use stable download URLs (#1916)
* ViTSmall16_Weights: fix typo (#1904)

## Samplers

* RandomGeoSampler: optional length is optional (#1907)

## Trainers

* Remove unnecessary argmax before call to torchmetrics (#1777)
* Better document default trainer metrics (#1874, #1914, #1923, #1924)
* ObjectDetectionTask: increase test coverage (#1739)

## Scripts

* SSL4EO download: skip downloading missing coordinates (#1821)
* Ensure that all files have the license header at the top (#1787)

## Tests

* Notebooks: use stable dependency versions (#1838)
* Don't cast warnings to errors (#1793)
* Fix lightning-utilities deprecation warning (#1733)
* Fix pre-commit dependency versions (#1781)

## Documentation

* RasterDataset: clarify documentation of is_image and dtype (#1811)
* RtD: use stable dependency versions (#1827)
* Document TorchGeo alternatives (#1742)
* Tutorials: load_state_dict does not return the model (#1503)
* README: fix VHR-10 example (#1686, #1920)
* README: add TorchGeo podcast episodes (#1806)
* README: add PyTorch badge (#1882)
* README: add OSGeo badge (#1880)
* README: add color lexing of bibtex (#1820)
* README: fix Spack link (#1804)

## Contributors

This release is thanks to the following contributors:

@adamjstewart
@ashnair1
@calebrob6
@DimitrisMantas 
@dmeaux 
@isaaccorley 
@jdilger
@julien-blanchon 
@konstantinklemmer 
@nilsleh
@tatsubori
